source/compiler/sc1.c

```diff
@@ -4048,9 +4048,8 @@ static int argcompare(arginfo *a1,arginfo *a2)
         result= a1->defvalue.array.size==a2->defvalue.array.size;
       if (result)
         result= a1->defvalue.array.arraysize==a2->defvalue.array.arraysize;
-      /* ??? should also check contents of the default array (these troubles
-       * go away in a 2-pass compiler that forbids double declarations, but
-       * Pawn currently does not forbid them) */
```

Comment:
I am unable to understand this comment. What were the original authors trying to say here?

Reply:
By "double declarations" the original author probably meant this:
```Pawn
forward Func(); // declared
forward Func(); // declared again, not an error
forward Func(arg); // error 025: function heading differs from prototype
```
Pawn allows functions to be forward-declared multiple times, as long as all declarations are identical.

Reply:
That's actually quite essential to some code.  There's a technique that actually abuses this and the definition of "identical" just being semantic.  These two are the same:

```pawn
Func(x[_:A:B:C:0]);
Func(x[_:D:E:F:0]);
```

So that code will compile with no issues, and is thus, as with all weird code that compiles, abused...